### PR TITLE
Kernel: Add a GPIO class for aarch64

### PR DESCRIPTION
**Pull request by nico**
This allows configuring the alternate pin functions and pin
pull up/down states, which is needed for using the UART.

### CODE REVIEW

File: Kernel/Prekernel/Arch/aarch64/GPIO.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ */
+
+#include <Kernel/Prekernel/Arch/aarch64/GPIO.h>
+#include <Kernel/Prekernel/Arch/aarch64/MMIO.h>
+
+extern "C" void wait_cycles(int n);
+
+namespace Prekernel {
+
+// See BCM2835-ARM-Peripherals.pdf section "6 General Purpuse I/O".
```
Comment (linusg) [inline Kernel/Prekernel/Arch/aarch64/GPIO.cpp:14]:
```suggestion
// See BCM2835-ARM-Peripherals.pdf section "6 General Purpose I/O (GPIO)".
```

They spell it correctly, I checked :^)

File: Kernel/Prekernel/Arch/aarch64/GPIO.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#pragma once
+
+#include <AK/Array.h>
+#include <AK/Types.h>
+
+namespace Prekernel {
+
+struct GPIOControlRegisters;
+
+// Can configure the general-purpose I/O registers on an Raspberry Pi.
```
Comment (linusg) [inline Kernel/Prekernel/Arch/aarch64/GPIO.h:16]:
```suggestion
// Can configure the general-purpose I/O registers on a Raspberry Pi.
```

Comment (nico) [reply Kernel/Prekernel/Arch/aarch64/GPIO.h:16]: (I had "RPi" first, but then forgot to update the article.)

**Comment by nico** [comment]
Thanks, all done.

**Pull request merged**